### DISCUSSION TITLE
Licence terms for website seem rather tangled?

### DISCUSSION THREAD

**Discussion by gilgongo** [jamulussoftware/jamulus#1872] [Documentation]
I've just noticed we have a sort of DIY [licence for the website repo](https://github.com/jamulussoftware/jamuluswebsite/blob/release/LICENSE.md) that says:  

"The source code of this repo doesn't have a specific license yet, but you are free to contribute, fork, modify ...." 

I think what Github expects is have a licence for the **code** here, not the content that's maintained with it. 

===

**EDIT**: Ignore the text below , which contains misconceptions on my part:

_Jekyll and po4a are MIT and GPL respectively, so that would mean putting it under GPL3. FoxCSS ~~seems to be un-licensed~~ CC BY. We state the licence to the **content** on the website footer (CC BY-SA, which is a bit mean in my opinion. I would have had it CC0, but it's too late now) so we should not state that here in case it confuses things._

_So I'd propose simply switching that licence to GPL3._

_BTW we can't I don't think [switch to MIT](https://www.tawesoft.co.uk/kb/article/mit-license-faq) if we have anything in the repo that's GPL, so our use of Jekyll prohibits that unless we're shelling out to it from the repo, is that right? In which case MIT is fine._

**Comment by ann0see** [comment]
If you look at the commit history, you should see that I added the license. 

Personally I don’t like applying the GPL and prefer MIT (especially due to copy left)

**Reply by hoffie** [reply]
> Personally I don’t like applying the GPL and prefer MIT (especially die to copy left)

I guess it depends a bit on how much "protection" we require regarding the website code. I don't think it contains any rocket science (maybe except for the po4a scripting? ;)), so I guess MIT would be OK? GPL would be in line with the Jamulus main repo though.

**Reply by ann0see** [reply]
GPL would be more in line, but after the missing clarity of the AppStore releases I would rather use MIT or similar which is more flexible.

**Reply by gilgongo** [reply]
I don't think we have the option of putting it under MIT if we are using GPLed components:

"Can I use GPL-licensed code in my MIT-licensed project?
No. The project as a whole must conform to the terms of the GPL license and must be distributed under the terms of that license. Therefore such a project as a whole must be distributed as GPL, but can still contain MIT-licensed software."  https://www.tawesoft.co.uk/kb/article/mit-license-faq

**Reply by ann0see** [reply]
But it depends on which code we’re referring to probably? We aren’t using po4a source code as far as I know and only fire a command to po4a in a bash script.

**Reply by gilgongo** [reply]
> We aren’t using po4a source code as far as I know and only fire a command to po4a in a bash script.

Oh wait - so po4a doesn't run from the repo that's on your machine? In which case that's OK since (hooray!) they aren't AGPL (if that's relevant).

~~So - we could put the website repo under MIT in that case! Just means we won't be able to use any GPL software in it in future. But what's the chances of that?~~ Our use of Jekyll still means we can't put the repo under MIT though.

**Comment by hoffie** [comment]
> Jekyll and po4a are MIT and GPL respectively, so that would mean putting it under GPL3.

As far as I can tell, we just use these tools (i.e. call the respective scripts/binaries), so I don't think this limits us in the choice of licenses?
Shelling out to open source tools is perfectly fine from proprietary software, AFAIK. Closer ties (such as linking) are more relevant though.

> FoxCSS seems to be un-licensed.

I guess we are bundling it, so it may be more relevant.
The code [claims to be CC-BY-3](https://github.com/ronanlevesque/FOX-CSS/blob/master/fox.css#L3).

**Reply by gilgongo** [reply]
As I say, I've always understood MIT to be sort of no-go with GPL, as confirmed by the FAQ: https://www.tawesoft.co.uk/kb/article/mit-license-faq#can-i-use-gpl-licensed-code-in-my-mit-licensed-project

**Reply by ann0see** [reply]
Yeah. But you can’t say: since this OS is GPL licensed, you can’t run MIT licensed code on it.

**Reply by gilgongo** [reply]
Yes, which is why it says  "...can still contain MIT-licensed software". So our use of Jeckyl is fine under GPL, I assume. 

EDIT: Sorry, I think I misunderstood you. This is about distributing the repo, not about running arbitrary things from it. the "OS" analogy doesn't fit.
